src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs

```diff
@@ -0,0 +1,272 @@
+            { WellKnownTags.Enum, CompletionItemKind.Enum },
+            { WellKnownTags.EnumMember, CompletionItemKind.EnumMember },
+            { WellKnownTags.Event, CompletionItemKind.Event },
+            { WellKnownTags.ExtensionMethod, CompletionItemKind.Method },
+            { WellKnownTags.Field, CompletionItemKind.Field },
+            { WellKnownTags.Interface, CompletionItemKind.Interface },
+            { WellKnownTags.Intrinsic, CompletionItemKind.Text },
+            { WellKnownTags.Keyword, CompletionItemKind.Keyword },
+            { WellKnownTags.Label, CompletionItemKind.Text },
+            { WellKnownTags.Local, CompletionItemKind.Variable },
+            { WellKnownTags.Namespace, CompletionItemKind.Text },
+            { WellKnownTags.Method, CompletionItemKind.Method },
+            { WellKnownTags.Module, CompletionItemKind.Module },
+            { WellKnownTags.Operator, CompletionItemKind.Operator },
+            { WellKnownTags.Parameter, CompletionItemKind.Value },
+            { WellKnownTags.Property, CompletionItemKind.Property },
+            { WellKnownTags.RangeVariable, CompletionItemKind.Variable },
+            { WellKnownTags.Reference, CompletionItemKind.Reference },
+            { WellKnownTags.Structure, CompletionItemKind.Struct },
+            { WellKnownTags.TypeParameter, CompletionItemKind.TypeParameter },
+            { WellKnownTags.Snippet, CompletionItemKind.Snippet },
+            { WellKnownTags.Error, CompletionItemKind.Text },
+            { WellKnownTags.Warning, CompletionItemKind.Text },
+        };
+
+        private readonly OmniSharpWorkspace _workspace;
+        private readonly FormattingOptions _formattingOptions;
+        private readonly ILogger _logger;
+
+        private (CSharpCompletionList Completions, string FileName, int Position)? _lastCompletion = null;
+
+        [ImportingConstructor]
+        public CompletionService(OmniSharpWorkspace workspace, FormattingOptions formattingOptions, ILoggerFactory loggerFactory)
+        {
+            _workspace = workspace;
+            _formattingOptions = formattingOptions;
+            _logger = loggerFactory.CreateLogger<CompletionService>();
+        }
+
+        public async Task<CompletionResponse> Handle(CompletionRequest request)
+        {
+            _logger.LogTrace("Completions requested");
+            _lastCompletion = null;
+
+            var document = _workspace.GetDocument(request.FileName);
+            if (document is null)
+            {
+                _logger.LogInformation("Could not find document for file {0}", request.FileName);
+                return new CompletionResponse { Items = ImmutableArray<CompletionItem>.Empty };
+            }
+
+            var sourceText = await document.GetTextAsync();
+            var position = sourceText.GetTextPosition(request);
+
+            var completionService = CSharpCompletionService.GetService(document);
```

Comment:
do you want to activate ShowItemsFromUnimportedNamespaces here?

Reply:
sorry, I read the PR description again and you explicitly mentioned it's not implemented yet.

IMHO this is a good idea to leave these extra things out and keep this PR manageable and merge, and only afterwards add extra features in separate PRs, if it makes sense